### ISSUE TITLE
Demotronic (GBC demo) has various glitches

### ISSUE THREAD

**Issue by waddlesplash** [mgba-emu/mgba#366]
http://www.pouet.net/prod.php?which=7175

mGBA 0083fad, Windows 10
What I noticed:
 - [ ] Colored squares with text (~0:40 in video) flashes black, doesn't do that in the video or in BGB.
 - [ ] Squares jitter between being misaligned to aligned again in mGBA (see ~1:05 in video for where it's most obvious, although that's happening the whole time -- BGB has no jitter)
 - [ ] Black borders on colored squares part that get larger and smaller (same deal as above for time/obviousness, also does not happen in BGB)
 - [ ] Never reaches the rest of the demo that starts at ~2:00 in the video. BGB does not reach it either. IT does have anti-emulation code, but that's on the title screen (it detects VBA-M at least...), and I don't have a flashcart so I can't test on real hardware.

**Comment by taisel** [comment]
The anti-emulation is where it reads off WAVE RAM. The end part that loops streams updates to horizontal and vertical scroll registers during scanline drawing intentionally.

**Comment by endrift** [comment]
d5cfcf219ec1502cc152b331e74e512c5c0af674 fixes these issues, except for the last scene, which requires implementing a scanline timing feature that I haven't implemented yet.

**Comment by taisel** [comment]
Yeah it's real fucky as it's mid-scanline drawing changes. The updates change the tile fetch patterns and the way it aliases to which tile the updates apply to depends on the I/o register setup. 

In other words: requires a parallel to accuracy profile bsnes for GBC.

**Comment by taisel** [comment]
In particular, an update happening at the same clock every line will affect the screen at a different starting offset into draw if things like hscroll (lower 3 bits) change.

**Comment by waddlesplash** [comment]
Added that and two more glitches I noticed to the list.

**Comment by endrift** [comment]
At this point, I'm going to leave PPU timing alone until 0.6, since getting into the nitty-gritty is really annoying, especially without a great test suite (mooneye-gb's suite is rather lackluster for actually figuring out PPU timings, although it can let you know some basic stuff, cc @gekkio). Better documentation would be ideal, and I've been trying to reverse that out of gambatte's PPU code.

**Comment by waddlesplash** [comment]
I think the scrolling text issue is actually fixed, as far as I can tell, so this one might be closeable.

**Comment by endrift** [comment]
Last time I checked that issue was still present, but I can check again in a little while.
